Zend/zend_builtin_functions.stub.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+function zend_version(): string {}
+
+function func_num_args(): int {}
+
+/** @return mixed */
+function func_get_arg(int $arg_num) {}
+
+/** @return array|false */
+function func_get_args() {}
+
+function strlen(string $str): int {}
```

Comment:
Generating macros to parse the arguments into zend_string/zend_long would be another way this could make code generation easier (e.g. `#define PARSE_ARGS_FOR_STRLEN(str) BEGIN....END` (possibly also declaring the variables)